### PR TITLE
[REVIEW] switch from CNMeM to `pool_memory_resource` [skip ci]

### PR DESCRIPTION
**Pull request by rongou**
Depends on https://github.com/rapidsai/rmm/pull/425.

@harrism @revans2 @jlowe

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@
 
 ## Improvements
 
+- PR #5632 Switch JNI code to use `pool_memory_resource` from CNMeM
```
Comment (harrism) [inline CHANGELOG.md:38]: "instead of cnmem", not "from cnmem".

Comment (rongou) [reply CHANGELOG.md]: Done.

**Comment by rongou** [comment]
> `java/README.md` needs to be updated. It refers to dynamically linking against librmm which is no longer the case after this change. It also has instructions for rebuilding RMM for static linking and/or for per-thread default stream.

Done.

**Pull request merged**